src/cryptonotecore/Core.cpp

```diff
@@ -2326,20 +2318,27 @@ namespace CryptoNote
             }
         }
 
-        if (block.timestamp > getAdjustedTime() + currency.blockFutureTimeLimit(previousBlockIndex + 1))
-        {
-            return error::BlockValidationError::TIMESTAMP_TOO_FAR_IN_FUTURE;
-        }
+	uint64_t futureTimeLimit;
 
-        auto timestamps = cache->getLastTimestamps(
-            currency.timestampCheckWindow(previousBlockIndex + 1), previousBlockIndex, addGenesisBlock);
-        if (timestamps.size() >= currency.timestampCheckWindow(previousBlockIndex + 1))
-        {
-            auto median_ts = Common::medianValue(timestamps);
-            if (block.timestamp < median_ts)
-            {
-                return error::BlockValidationError::TIMESTAMP_TOO_FAR_IN_PAST;
-            }
+	if (previousBlockIndex + 1 >= CryptoNote::parameters::DIFFICULTY_TARGET_V2_HEIGHT)
+	{
+		futureTimeLimit = CryptoNote::parameters::DIFFICULTY_TARGET * 6;
+	}
+	else
+	{
+		futureTimeLimit = CryptoNote::parameters::DIFFICULTY_TARGET_V2 * 6;
```

Comment:
These two statements are the wrong way round - this line should be `CryptoNote::parameters::DIFFICULTY_TARGET * 6` and visa versa.

Also, there is a variable `CRYPTONOTE_BLOCK_FUTURE_TIME_LIMIT` [here](https://github.com/derogold/derogold/blob/rebase-trtl-v18/src/config/CryptoNoteConfig.h#L39) which could be used instead of using the `6` magic number. Could add a `CRYPTONOTE_BLOCK_FUTURE_TIME_LIMIT_V2` if you liked, since you don't have that yet.

Reply:
makes sense, ty. the only thing where I'm a bit confused - I'm looking at the CRYPTONOTE_BLOCK_FUTURE_TIME_LIMIT variable in the cryptonoteconfig and I have it define as 
const uint64_t CRYPTONOTE_BLOCK_FUTURE_TIME_LIMIT = 6 * DIFFICULTY_TARGET_V2;

shouldn't it be 6 * DIFFICULTY_TARGET here? 

and the when I would define the V2 it would be here?
const uint64_t CRYPTONOTE_BLOCK_FUTURE_TIME_LIMIT_V2 = 6 * DIFFICULTY_TARGET_V2;

Reply:
I think this might be another case of my initial changes making the code confusing - instead of having a v0/v1/v2/v3 like in the TurtleCoin codebase, we just have the original value for future time limit, and at some point it looks like that was not updated to account for the new block time.

Searching through the `git blame`, looks like that was actually introduced by me, here: https://github.com/derogold/derogold/commit/bd2add816d4e4c90bb8c5d8c37f3a0cf72dfcbf7

I guess, given that it's been like this for so long then those future time limits are in face the 'correct' ones, since that's what the chain has been using?